vite.config.mts

```diff
@@ -0,0 +1,172 @@
+import react from '@vitejs/plugin-react-swc'
+import * as fs from 'fs'
+import { CID } from 'multiformats/cid'
+import * as raw from 'multiformats/codecs/raw'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { dirname, resolve } from 'path'
+import * as path from 'path'
+import * as ssri from 'ssri'
+import { fileURLToPath } from 'url'
+import type { PluginOption } from 'vite'
+import { defineConfig } from 'vite'
+import { analyzer } from 'vite-bundle-analyzer'
+import checker from 'vite-plugin-checker'
+import { nodePolyfills } from 'vite-plugin-node-polyfills'
+import tsconfigPaths from 'vite-tsconfig-paths'
+
+import { cspMeta, headers, serializeCsp } from './headers'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+const VITE_CSP_META = serializeCsp(cspMeta)
+
+const publicFilesEnvVars = {
+  VITE_CSP_META: JSON.stringify(VITE_CSP_META),
+}
+
+const publicPath = path.join(__dirname, 'public')
+
+for (const dirent of fs.readdirSync(publicPath, { withFileTypes: true })) {
+  if (!dirent.isFile()) continue
+  const mungedName = dirent.name
+    .toUpperCase()
+    .split('')
+    .map(x => (/^[0-9A-Z]$/.test(x) ? x : '_'))
+    .join('')
+  const data = fs.readFileSync(path.join(publicPath, dirent.name))
+
+  const integrity = ssri.fromData(data, {
+    strict: true,
+    algorithms: ['sha256'],
+  })
+  publicFilesEnvVars[`VITE_SRI_${mungedName}`] = JSON.stringify(integrity)
+
+  const digest = sha256.digest(data)
+  const cid = CID.create(1, raw.code, digest).toString()
+  publicFilesEnvVars[`VITE_CID_${mungedName}`] = JSON.stringify(cid)
+}
+
+const defineGlobalThis: PluginOption = {
```

Comment:
Can we add a comment to have more context on this? I can imagine someone wondering why we have this in the future

Reply:
2a3f0b57095fe22d647d4ecf0ff77bf75940c962